### PR TITLE
Remove semantic files

### PR DESCRIPTION
**Pull request by ghost**
\#24 #45 

# Description
Removed generated semantic-ui output files from repo and updated readme.

# Change Log
- `config/webpack.config.dev.js` `config/webpack.config.prod.js` : allow import of modules outside of the src/ directory
- `semantic/` `views/semantic/` : deleted
- `semantic.json`: allow `autoInstall` to bypass interactive install 
- `Readme.md` : removed interactive install process
- `index.js` : new import location for semantic
- `.gitignore` : ignore generated semantic files, keep `semantic.json`
- `src/view/navbar/index.js`: removed unused modules, added `alt` prop to img
- `.travis.yml`: updated to reflect new build process

# Manual Test Steps
Clone repository and follow steps indicated in the readme, ensure that the page looks as it should:

screen shot 2017-10-02 at 1 40 04 pm

### CODE REVIEW

**Comment by noahpoz** [comment]
I think the Travis build fails because the install process for the semantic-ui module is interactive.

**Comment by gjbunyea** [comment]
@noahpoz When I run `npm install` on your branch I don't get any interactive stuff for semantic and have no `semantic/` folder

**Comment by gjbunyea** [comment]
My 10 (:-1:) commits fix #45

**Review by gjbunyea — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -65,6 +66,7 @@ dist/# See https://help.github.com/ignore-files/ for more about ignoring files.
 
 # dependencies
 /node_modules
+# semantic.json
```
Comment (angushtlam) [inline .gitignore]: Why does the changelog say the `semantic.json` is deleted but you're including it still in the repo? So what exactly is happening with this file?

File: README.md [hunk 2]
```diff
@@ -10,9 +10,23 @@ To view our prototype, clone and enter the following repo:
 git clone git@github.com:project-ivi/ivi.git
 cd ivi
 ```
-Then install and build the sources.
+Install the dependencies:
 ```
 npm install
+```
+During the installation process, some additional steps are necessary to install the semantic-ui module. Navigate the prompts as follows:
+- 'Set-up Semantic UI' > Select `Automatic`
```
Comment (angushtlam) [inline README.md]: I thought this would be automated with no CLI input when `autoInstall` is true in `semantic.json`?

File: semantic.json [hunk 3]
```diff
@@ -8,66 +8,15 @@
...
+      "compressed": "dist/components/",
+      "themes": "dist/themes/"
     },
     "clean": "dist/"
   },
   "permission": false,
-  "autoInstall": false,
+  "autoInstall": true,
   "rtl": false,
-  "components": [
```
Comment (angushtlam) [inline semantic.json]: Why are these components gone from the file now? Don't we use some of these?

File: src/view/navbar/index.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const Navbar = () => (
     <Menu className='component__navbar'>
       <span style={{width: '100%', height: '100%', display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
         <span style={{height: '100%', padding: '8px'}}>
-          <img src={Logo} className='ivy-logo'/>
+          <img alt='' src={Logo} className='ivy-logo'/>
```
Comment (angushtlam) [inline src/view/navbar/index.js]: Alt tag is empty?

Comment (gjbunyea) [reply src/view/navbar/index.js]: fixed.

File: README.md (hunk 2 above)
Comment (gjbunyea) [reply README.md]: good catch - i removed this.

File: .gitignore (hunk 1 above)
Comment (gjbunyea) [reply .gitignore]: Updated the changelog - we need `semantic.json` to get the autoInstall

File: semantic.json (hunk 3 above)
Comment (gjbunyea) [reply semantic.json]: I think this is what was trying to accomplish with this PR. Is that accurate @noahpoz ?

Comment (noahpoz) [reply semantic.json]: Yes. The way I originally configured it was cherry picking the desired components (which ended up being all of them anyway). That's what's going on here. Specifying the components like this isn't necessary when autoInstall is true.

File: .gitignore (hunk 1 above)
Comment (noahpoz) [reply .gitignore]: I originally deleted semantic.json to enable the interactive installer, but Greg noticed afterward that we can enable autoInstall in this file, so we actually need to keep it.

File: semantic.json (hunk 3 above)
Comment (angushtlam) [reply semantic.json]: Sweet!

File: .gitignore (hunk 1 above)
Comment (angushtlam) [reply .gitignore]: 👍

**Review by angushtlam — APPROVED** [review]
lg2m

**Pull request merged**